### PR TITLE
fix: ensure OSD is sublcassable

### LINKED ISSUE #312: zope.interface 7 causing breakage
In an unrelated project I am seeing the traceback shown below during GHA test runs using tox and zope.testrunner. The mentioned `_zope_interface_coptimizations.ObjectSpecificationDescriptor` is an addition in zope.interface version 7. I had to pin zope.interface to the previous release to make it go away.

```
py311-gha: commands[1]> coverage run /home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/bin/test
Traceback (most recent call last):
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/bin/test", line 41, in <module>
    sys.exit(zope.testrunner.run((['-cv']) + [
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/testrunner/__init__.py", line 31, in run
    failed = run_internal(defaults, args, script_parts=script_parts, cwd=cwd,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/testrunner/__init__.py", line 52, in run_internal
    from zope.testrunner.runner import Runner
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/testrunner/runner.py", line 48, in <module>
    import zope.testrunner.tb_format
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/testrunner/tb_format.py", line 20, in <module>
    import zope.exceptions.exceptionformatter
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/exceptions/__init__.py", line 37, in <module>
    import zope.security  # noqa: suppress unused import warning from flake8
    ^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/security/__init__.py", line 20, in <module>
    import zope.security.decorator
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/security/decorator.py", line 22, in <module>
    from zope.proxy.decorator import SpecificationDecoratorBase
  File "/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/zope/proxy/decorator.py", line 30, in <module>
    class DecoratorSpecificationDescriptor(ObjectSpecificationDescriptor):
TypeError: type '_zope_interface_coptimizations.ObjectSpecificationDescriptor' is not an acceptable base type
/home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/lib/python3.11/site-packages/coverage/control.py:894: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
py311-gha: exit 1 (0.56 seconds) /home/runner/work/mw.importlib/mw.importlib> coverage run /home/runner/work/mw.importlib/mw.importlib/.tox/py311-gha/bin/test pid=3627
  py311-gha: FAIL code 1 (103.80=setup[36.75]+cmd[66.49,0.56] seconds)
  evaluation failed :( (104.00 seconds)
```

### PR DESCRIPTION
**Pull request by tseaver**
Broken in 7.0 release.

Closes #312.

### CODE REVIEW

**Comment by tseaver** [comment]
Note to reviewers: the new `SubclassableMixin` adds a test which failed (for the C-based `ObjectSpecificationDescriptor`) before the change to the C extension.

**Review by dataflake — APPROVED** [review]
I can't comment on the C bits, but thanks for the quick turnaround!

File: CHANGES.rst [hunk 1]
```diff
@@ -2,10 +2,11 @@
  Changes
 =========
 
-7.1 (unreleased)
-================
+7.0.post0 (unreleased)
```
[the remark this answers is not in the document]
Comment (tseaver) [reply CHANGES.rst:5]: @davisagli 
> I'd call that 7.0.1?
That was my original thought, but we seem to have shifted in the recent past to use only `x.y` releases -- I would call this one an "emergency fix".

[the remark this answers is not in the document]
Comment (davisagli) [reply CHANGES.rst]: I'm fine with either. I'm not familiar with the reasoning for using the post suffix for this package.

[the remark this answers is not in the document]
Comment (tseaver) [reply CHANGES.rst]: 19774cd

**Review by davisagli — APPROVED** [review]

**Pull request merged**